templates/helm/templates/deployment.yaml

```
@@ -141,8 +141,15 @@ spec:
       priorityClassName: {{ .Values.deployment.priorityClassName }}
       {{ end -}}
       hostIPC: false
-      hostNetwork: false
       hostPID: false
+      {{ if .Values.deployment.hostNetwork }}
+      hostNetwork: true
+      dnsPolicy: ClusterFirstWithHostNet
+      {{ else }}
+      hostNetwork: false
+      {{ with .Values.deployment.dnsPolicy }}
+      dnsPolicy: {{ . }}
+      {{ end }}
```

Based on the error logs from the prow job, it appears that there may be some formatting errors. I recommend generating a controller and testing it to verify if it functions correctly.

---

(not on a laptop right now, i can try to generate and give more concise feedback tomorrow)

---

not familiar with the `with` key word here, let's keep it simple and do something like
```suggestion
      {{ if .Values.deployment.hostNetwork }}
      hostNetwork: true
      dnsPolicy: ClusterFirstWithHostNet
      {{ else }}
      hostNetwork: false
      dnsPolicy: {{ Values.deployment.dnsPolicy }}
      {{ end }}
```

---

or maybe something like:
```yaml
      hostNetwork: {{ .Values.deployment.hostNetwork }}
      dnsPolicy: {{ .Values.deployment.dnsPolicy }}
```

---

Yes, that is true. I set a default value and it works, so that's what I'll do.

---

Thx